docs/api/modulus.metrics.rst

```diff
@@ -1,9 +1,266 @@
+    >>> y = torch.randn((1,))
+    >>> bins, counts = histogram.histogram(x, bins = 10)
+    >>> ranks = calibration.find_rank(bins, counts, y)
+    tensor([0.1920])
+
+Wasserstein Metric:
+
+.. code:: python 
+
+    >>> from modulus.metrics.general import wasserstein, histogram
+    >>> x = torch.randn((1_000,1))
+    >>> y = torch.randn((1_000,1))
+    >>> bins, cdf_x = histogram.cdf(x)
+    >>> bins, cdf_y = histogram.cdf(y, bins = bins)
+    >>> wasserstein(bins, cdf_x, cdf_y)
+    >>> wasserstein.wasserstein(bins, cdf_x, cdf_y)
+    tensor([0.0459])
+  
+
+Weighted Reductions
+^^^^^^^^^^^^^^^^^^^
+Modulus currently offers classes for weighted mean and variance reductions.
+
+.. code:: python
+
+    >>> from modulus.metrics.general import reduction
+    >>> x = torch.randn((1_000,))
+    >>> weights = torch.cos(torch.linspace(-torch.pi/4, torch.pi/4, 1_000))
+    >>> wm = reduction.WeightedMean(weights)
+    >>> wm(x, dim = 0)
+    tensor(0.0365)
+    >>> wv = reduction.WeightedVariance(weights)
+    >>> wv(x, dim = 0)
+    tensor(1.0148)
+
+
+Online Statistical Methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Modulus current offers routines for computing online, or out-of-memory, means,
+variances, and histograms.
+
+.. code:: python 
+
+  >>> import torch
+  >>> from modulus.metrics.general import ensemble_metrics as em
+  >>> x = torch.randn((1_000, 2)) # Interpret as 1_000 members of size (2,).
+  >>> torch.mean(x, dim = 0) # Compute mean of entire data.
+  tensor([-0.0545,  0.0267])
+  >>> x0, x1 = x[:500], x[500:] # Split data into two.
+  >>> M = em.Mean(input_shape = (2,)) # Must pass shape of data
+  >>> M(x0) # Compute mean of initial batch.
+  tensor([-0.0722,  0.0414])
+  >>> M.update(x1) # Update with second batch.
+  tensor([-0.0545,  0.0267])
+
+
+Climate Related Metrics
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To compute the Anomaly Correlation Coefficient, a metric widely used in weather and
+climate sciences:
+
+.. code:: python
+
+    >>> import torch
+    >>> import numpy as np
+    >>> from modulus.metrics.climate.acc import acc
+    >>> channels = 1
+    >>> img_shape = (32, 64)
+    >>> time_means = np.pi / 2 * np.ones((channels, img_shape[0], img_shape[1]), dtype=np.float32)
+    >>> x = np.linspace(-180, 180, img_shape[1], dtype=np.float32)
+    >>> y = np.linspace(-90, 90, img_shape[0], dtype=np.float32)
+    >>> xv, yv = np.meshgrid(x, y)
+    >>> pred_tensor_np = np.cos(2 * np.pi * yv / (180))
+    >>> targ_tensor_np = np.cos(np.pi * yv / (180))
+    >>> pred_tensor = torch.from_numpy(pred_tensor_np).expand(channels, -1, -1)
+    >>> targ_tensor = torch.from_numpy(targ_tensor_np).expand(channels, -1, -1)
+    >>> means_tensor = torch.from_numpy(time_means)
+    >>> lat = torch.from_numpy(y)
+    >>> acc(pred_tensor, targ_tensor, means_tensor, lat)
+    tensor([0.9841])
+
+
 .. autosummary::
    :toctree: generated
 
+General
+---------
+
+.. automodule:: modulus.metrics.general.mse
+    :members:
+    :show-inheritance:
+
+.. _histogram:
```

Comment:
Is this suppose to be here?

Reply:
Great catch. I was trying something and this was a leftover from the testing. Removed